pkg/util/ipset/ipset.go

```diff
@@ -91,36 +92,44 @@ type IPSet struct {
 	Comment string
 }
 
+var (
+	errPortNotSpecified = errors.New("should specify right port range for bitmap type set")
+	errWrongSizeNumber  = errors.New("wrong hash size number")
+	errWrongHashNumber  = errors.New("wrong hash max elem number")
+	errWrongProtocol    = errors.New("wrong protocol")
+	errWrongSetType     = errors.New("wrong set type")
+)
+
 // Validate checks if a given ipset is valid or not.
-func (set *IPSet) Validate() bool {
+func (set *IPSet) Validate() error {
 	// Check if protocol is valid for `HashIPPort`, `HashIPPortIP` and `HashIPPortNet` type set.
 	if set.SetType == HashIPPort || set.SetType == HashIPPortIP || set.SetType == HashIPPortNet {
 		if valid := validateHashFamily(set.HashFamily); !valid {
```

Comment:
I'd prefer this push all the way down:

```
		if err := validateHashFamily(set.HashFamily); err != nil {
			return err
		}
	}
```

Then functions like `validateHashFamily` don't need to log (util libs generally shouldn't), they just need an error, and the definition of that erro ris very localized.

Reply:
leaving `validateHashFamily` like this? 
```go
// checks if given hash family is supported in ipset
func validateHashFamily(family string) error {
	if family == ProtocolFamilyIPV4 || family == ProtocolFamilyIPV6 {
		return nil
	}
	return fmt.Errorf("Currently supported ip set hash families are: [%s, %s], %s is not supported", ProtocolFamilyIPV4, ProtocolFamilyIPV6, family)
}
```
??

Reply:
Thanks for your prompt review :)

Reply:
Yes, but the error can be more terse: `"unuspported IP family: %s"`

Reply:
addressed requested changes